### DISCUSSION TITLE
Improve category names in Supported Manager docs

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#23174] [Suggest an Idea]
### Background info

@secustor added categories to the code in these PRs:

- #16534
- #23162

I like the new categories. It's easy to see the relevant section for your programming language! 🥳 

I have some style suggestions to make the list even better. 🙃 I don't know if this is hard/easy to change in the code. Let me know what you think of my ideas at least.

### Expand abbreviations

The published docs has category names like **cd** and **ci**, which are a bit hard to understand. [^1] I would like to see **Continuous Delivery** instead of **cd** for example.

### Match upstream capitalization

I'd also like to match the programming language's capitalization, like JavaScript, PHP, Python, Ruby, Rust, and so on.

### Replace `-` with ` `

And I'd like to replace `-` with ` `.

### Summary

Here's a short summary of the kind of changes I'd like. The table does not show all entries!

| Current category name | New displayed category |
| --- | --- |
| **cd** | **Continuous Delivery** |
| **ci** | **Continuous Integration** |
| **iac** | **Infrastructure as Code** |
| **js** | **JavaScript** |
| **node** | **Node.js** |
| **php** | **PHP** |
| **no-category** | **no category** |

[^1]: [Renovate docs, supported managers](https://docs.renovatebot.com/modules/manager/#supported-managers)

**Comment by secustor** [comment]
I agree that is not intuitive for looking up stuff, but we have to show both the displayName and the actual id as the later is used for configuration. 

Implementation wise that shouldn't be problem.

**Reply by HonkingGoose** [reply]
> I agree that is not intuitive for looking up stuff, but we have to show both the displayName and the actual id as the later is used for configuration.

Listing the display name _and_ the ID will be confusing. So let's only show the actual ID. I guess whitespace like ` ` is not allowed in the ID?

Following upstream capitalization or style makes the ID confusing: "Should I use `php` or `PHP`, `Node.js` or `node`, and so on.

I guess that only leaves expanding abbreviations like `cd`, `ci` and `iac` as something we can/should do? 😄 

> Implementation wise that shouldn't be problem.

Okay! I guess changing the ID is a _breaking change_?

**Reply by secustor** [reply]
I have been thinking of a table like this instead of https://docs.renovatebot.com/modules/manager/#supported-managers 

| Name | id | managers |
|--------|--------|--------|
| Node.js | `node` | `nodeenv`, `nvm`  |
| Continous Integration | `ci` | `azure-pipelines`, `gitlabci`, ... | 

In code there will be 1:1 mapping between id and displayName.

>  >Implementation wise that shouldn't be problem.

> Okay! I guess changing the ID is a breaking change?

Yes

**Reply by HonkingGoose** [reply]
I love that table! Do you want me to create an issue for the table idea, and assign you to it?

**Reply by secustor** [reply]
Please, but before that can you create a table here, so we can align on the display names first?

**Reply by HonkingGoose** [reply]
Hi @secustor,

Here's my table:

| Name                       | ID            | Managers        |
| -------------------------- | ------------- | --------------- |
| **Ansible**                | `ansible`     | same as current |
| **Batect**                 | `batect`      | same as current |
| **Bazel**                  | `bazel`       | same as current |
| **C and C++**              | `conan`       | same as current |
| **Continuous Delivery**    | `cd`          | same as current |
| **Continuous Integration** | `ci`          | same as current |
| **Dart**                   | `dart`        | same as current |
| **Docker**                 | `docker`      | same as current |
| **.NET**                   | `dotnet`      | same as current |
| **Elixir**                 | `elixir`      | same as current |
| **Go**                     | `golang`      | same as current |
| **Helm**                   | `helm`        | same as current |
| **Infrastructure as Code** | `iac`         | same as current |
| **Java**                   | `java`        | same as current |
| **JavaScript**             | `js`          | same as current |
| **Kubernetes**             | `kubernetes`  | same as current |
| **Node.js**                | `node`        | same as current |
| **PHP**                    | `php`         | same as current |
| **Python**                 | `python`      | same as current |
| **Ruby**                   | `ruby`        | same as current |
| **Rust**                   | `rust`        | same as current |
| **Swift**                  | `swift`       | same as current |
| **Terraform**              | `terraform`   | same as current |
| **No category**            | `no-category` | same as current |

It would be nice to have the ID match the display name, but that's a breaking change. Also: some IDs may not be valid TypeScript variables, like `C-and-C-++` or `C-and-C-plus-plus`, or `.NET`. 😄

**Reply by secustor** [reply]
Thanks, LGTM!

Tough WDYT about capitalizing `integration` and `delivery`? 
Looks wrong in combination with a capitalized `Code` of `Infrastructure as Code`.

**Reply by HonkingGoose** [reply]
Capitalizing `Integration` and `Delivery` is fine with me. 😄 Do you want me to make a issue now then?

**Reply by secustor** [reply]
Yes, please :)

**Reply by HonkingGoose** [reply]
See issue:

- #23233
